mne/preprocessing/eyetracking/eyetracking.py

```diff
@@ -5,7 +5,100 @@
 
 import numpy as np
 
+from ...io import BaseRaw
 from ...io.constants import FIFF
+from ...utils import logger, _check_preload, _validate_type, warn
+
+
+def interpolate_blinks(raw, buffer=0.05, match="BAD_blink"):
```

Comment:
Should this also take channels? Is there any scenario where someone would like to interpolate for one channel but not the other?

Reply:
I think we included this in the "let's hold off for now and implement it if people ask for it". 

I think It's a little tough because for example if you only interpolate over some channels and not others then the data in the blink segments are still partly `"BAD"` (i.e. unusable samples or `np.nan` within the blink period).

I personally haven't had a scenario in my research where I've only wanted to interpolate some eyetrack channels and not others. What about you?

Reply:
That's fair! I think I also need to remind myself that this is only for pupillometry channels to begin with (right?). Which actually makes me wonder if we should name the function more specifically to mark that - `interpolate_blinks_pupil` or something? We do not foresee this function to take over other channels as well, right?

Reply:
I think the idea is that the `eyetracking` namespace it's nested under as `mne.preprocessing.eyetracking.interpolate_blinks` should be good enough to indicate the scope

Reply:
@britta-wstnr You are totally right this is what we initially were going to do, but @larsoner and I met last week and felt that it's okay to interpolate over both `eyegaze` and `pupil` channels : )

Reply:
Mhhh I am not sure I agree interpolating `eyegaze` is a good idea - the position of the eye on the screen does not exist during a blink.

Reply:
My argument was that the eye is still pointing somewhere in space and that's what the eye position/gaze is meant to reflect (even if it's blocked by an eyelid or something else). And it probably doesn't move during the blink / that is a reasonable assumption for most blinks.

Reply:
... I guess the question is what concrete use cases will be broken by interpolating? One that will be helped would be saying "the subject stayed fixed on a cross" or something so you can use `np.std` instead of `np.nanstd` etc. and it should still be valid. To me making people worry about `nan` when filtering, processing, etc. is not good for downstream stuff. My vote would be that the interpolation should interpolate everything by default at least, and we could add an option later to not do it if people end up needing it. But if that seems unsafe in some ways then we can do the reverse (though I'd like to add the `interpolate_gaze=False` *now* because I think I'd always want to set it to True in my analyses).

Reply:
Mh I don't think that is true - after all you can see blinks in EOG channels because the eye moves, right? -- A quick literature search indeed suggests that there is eye movement associated with blinking.
And especially if the gaze is expressed as position on the screen, an interpolation would be very misleading.

Reply:
> Mh I don't think that is true - after all you can see blinks in EOG channels because the eye moves, right? -- A quick literature search indeed suggests that there is eye movement associated with blinking.

I am with @britta-wstnr here, I thought that the eyes naturally roll upward during blinks

Reply:
> Part of the decision as I remember was one of practicality.

So maybe we do add `interpolate_gaze=False` and if people want to do something where they know the gaze is technically undefined but want `epochs.average` and `.filter(picks='gaze')` or whatever to "just work" then they can do it.

Reply:
I wonder if the averaging would be something to fix in the averaging instead of interpolating (for gaze, that is).
Regarding the annotations: could the annotations be only removed from the pupil channels then? If I envision research paired with MEG and EEG, you'd want to retain information about blink periods anyway, even if you might want to interpolate your pupil data. I think it is not uncommon to clean MEEG data based on eye tracking information (e.g., I often kick out all trials that have a blink event in the eye tracking in my visual processing studies).

Reply:
Alright sounds like the consensus is to not interpolate `eyegaze` channels by default -  I will add the `interpolate_gaze=False` flag and push a commit!


I agree that `np.nanmean` might be something we want to consider in the future for `epochs.average()` if this is an issue people keep running into!